docs/hotplug.md

```diff
@@ -0,0 +1,55 @@
+# Cloud Hypervisor Hot Plug
+
+Currently Cloud Hypervisor only support hot plugging of CPU devices.
+
+## Kernel support
+
+For hotplug on Cloud Hypervisor ACPI GED support is needed. This can either be achieved by turning on `CONFIG_ACPI_REDUCED_HARDWARE_ONLY` 
+or by using this kernel patch: https://git.kernel.org/pub/scm/linux/kernel/git/rafael/linux-pm.git/commit/?id=ac36d37e943635fc072e9d4f47e40a48fbcdb3f0
+
+This patch is integrated into the Clear Linux KVM and cloudguest images.
+
+## CPU Hot Plug
+
+Extra vCPUs can be added (but not removed) from a running Cloud Hypervisor instance. This is controlled by two mechanisms:
```

Comment:
Could we specify is CPU hot un-plug will ever be supported?

Reply:
It's an easy mechanism, and in the context of k8s pod scaling, we might need a VM to scale up and down, so I think we have to support it at some point.
Memory unplug is another story...

Reply:
Agreed. I think we may want to call out that we won't support ACPI based memory unplug.

Reply:
> Agreed. I think we may want to call out that we won't support ACPI based memory unplug.

Once we add a section about memory then we can talk about it then. We might use e.g. virtio-mem for memory which does offer better reclamation.

Reply:
Updated.

Reply:
>Once we add a section about memory then we can talk about it then. We might use e.g. virtio-mem for memory which does offer better reclamation.

Yes I agree. I think for now the ACPI memory hotplug will be the only thing we support, giving half of the flexibility, but we need something like virtio-mem to provide a fully flexible `unplug` solution.

@sameo @rbradford 
One idea about memory hotplug. Because `virtio-balloon` is very close to what `virtio-mem` is doing, maybe we could think about implementing `virtio-balloon` now, and the transition to `virtio-mem` will be easy to manage. This would provide us a way to unplug memory.